### ISSUE TITLE
Garbled display of Japanese in imported .jww files

### ISSUE THREAD

**Issue by zxq9** [LibreCAD/LibreCAD#150]
Display of existing Japanese in .jww files is garbled. (And maybe .dxf files also, haven't tested yet.)

This could be (in descending order of likelihood):

1- A missing font.
  Not sure what font package JwCAD uses yet.
2- An encoding issue.
  It is possible that JwCAD encodes in Shift_JIS or ISO-2022-JP or something similar.
3- A format translation issue.
  JwCAD may do any number of incompatible things when saving text -- but this is the least likely source of the problem.

**Comment by magnusroe** [comment]
Does anyone know anything about this? I work for an architect in Japan and every plan must be sent as a JWW file, it really doesn't play well with Autocad and I've never been able to find out how to export from Autocad to Jwcad without garbling (although I get files from clients that use autocad that work, I haven't been able to establish how).

**Comment by zxq9** [comment]
I haven't messed with this for a few years. Not sure the status right now. As far as I know Japanese text garbling is still an issue.

As far as Autocad -> Jw you will have to export an old version of DXF -- if that is even an option. DXF is sort of like MS Office formats in that there are many internal sub-versions that are often slightly incompatible. Jwcad can only read older versions of DXF, so if there is an option in Autocad to save as an older DXF version give it a shot -- I believe it was still reading DXF reasonably well around 2008. I don't remember if Japanese text worked between the two, though.

**Comment by fa201** [comment]
Is this issue still valid ?

**Comment by zxq9** [comment]
I believe so. I can test this coming week.

**Comment by magnusroe** [comment]
Yes the issue remains, I have managed to get around it by working in .dxf files in another app and using AR_Cad to save as jww.

**Comment by app4soft** [comment]
I tried to open JWW samples distributed with _Jw_cad_ `8.25a`, but latest _LirbeCAD_ `2.2.2_alpha` can't open it (it continuosly loading file after click to open it).

There is some specification on Jw_cad's file formats (is it for v7? or v8? its unclear) — can LibreCAD's `jwwlib` actually handle Jw_cad's v7 and v8 JWW files?

- http://www.jwcad.net/jwdatafmt.txt
  - https://github.com/LibreCAD/LibreCAD/tree/master/libraries/jwwlib

**Comment by siwf** [comment]
I opened some [JWW file][jww_samples], then check properties of text entity,
the text is not japanese, it should be text encoding issue.

[jww_samples]: https://github.com/monozukuri-ai/ezjww/tree/main/jww_samples
